File: rss-reader/app/src/main/java/com/abubusoft/kripton/examples/rssreader/model/Thumbnail.java
package com.abubusoft.kripton.examples.rssreader.model;

import com.abubusoft.kripton.annotation.BindType;
import com.abubusoft.kripton.annotation.BindXml;
import com.abubusoft.kripton.xml.XmlType;

import java.net.URL;

@BindType
public class Thumbnail {
    @BindXml(xmlType = XmlType.ATTRIBUTE)
    public long width;

    @BindXml(xmlType = XmlType.ATTRIBUTE)
    public long height;

    @BindXml(xmlType = XmlType.ATTRIBUTE)
    public URL url;
}

File: kotlin-rss-reader/app/src/main/java/com/abubusoft/kripton/example/rssreader/service/model/Thumbnail.kt
package com.abubusoft.kripton.example.rssreader.service.model

import com.abubusoft.kripton.annotation.BindType
import com.abubusoft.kripton.annotation.BindXml
import com.abubusoft.kripton.xml.XmlType
import java.net.URL

@BindType
class Thumbnail(
        @BindXml(xmlType = XmlType.ATTRIBUTE)
        val width: Long = 0,

        @BindXml(xmlType = XmlType.ATTRIBUTE)
        val height: Long = 0,

        @BindXml(xmlType = XmlType.ATTRIBUTE)
        val url: URL? = null
)
